nautobot/docs/plugins/development.md

```diff
@@ -1146,3 +1146,147 @@ try:
 except ImportError:
     pass
 ```
+
+## NautobotUIViewSet
+
+New in Nautobot 1.4 is the debut of `NautobotUIViewSet`: A powerful plugin development tool that can save plugin developer hundreds of lines of code compared to using legacy `generic.views`. Using it to gain access to default functionalities previous provided by `generic.views` such as `create()`, `bulk_create()`, `update()`, `partial_update()`, `bulk_update()`, `destroy()`, `bulk_destroy()`, `retrieve()` and `list()` actions.
+
+Note that this ViewSet is catered specifically to the UI, not the API.
+
+Concrete examples on how to use `NautobotUIViewSet` resides in `nautobot.circuits.views`.
+
+Below we provide an example on how to use `NautobotUIViewSet` on a theoretical plugin model.
+
+```python
+from nautobot.core.views.viewsets import NautobotUIViewset
+
+class YourPluginModelUIViewSet(NautobotUIViewSet):
+    bulk_create_form_class = YourPluginModelCSVForm
+    bulk_update_form_class = YourPluginModelBulkEditForm
+    filterset_class = YourPluginModelFilterSet
+    filterset_form_class = YourPluginModelFilterForm
+    form_class = YourPluginModelForm
+    queryset = YourPluginModel.objects.all()
+    serializer_class = serializers.YourPluginModelSerializer
+    table_class = YourPluginModelTable
+```
+
+### Setting ViewSet Attributes
+
+**One caveat of using the NautobotUIViewSet is that the `queryset`, `serializer_class` and `table_class` attribute of the `YourPluginModelUIViewSet` has to be set before most of the `NautobotUIViewSet` functionalities will become available.**
```

Comment:
Thinking about this, the `serializer_class` attribute would be nice to not require in the future, so that it's possible to use NautobotUIViewSet before implementing the REST API for a model.